kore/src/Kore/Step/SMT/Lemma.hs

```diff
@@ -72,41 +74,49 @@ declareSMTLemmas m = do
         -> m (Maybe ())
     declareRule (atts, axiomDeclaration) = runMaybeT $ do
         guard (isSmtLemma $ Attribute.smtLemma atts)
-        (lemma, vars) <-
+        (lemma, VarContext { terms }) <-
             runTranslator
             $ translatePredicate translateUninterpreted
             $ wrapPredicate $ sentenceAxiomPattern axiomDeclaration
-        SMT.assert (addQuantifiers vars lemma)
+        SMT.assert (addQuantifiers terms lemma)
 
     addQuantifiers vars lemma | null vars = lemma
     addQuantifiers vars lemma = SMT.List
         [ SMT.Atom "forall"
         , SMT.List
-            [ SMT.List [sexpr, t] | (sexpr, t) <- Map.elems vars ]
+            [ SMT.List [SMT.Atom smtName, smtType]
+            | SmtEncoding { smtName, smtType } <- Map.elems vars ]
         , lemma
         ]
 
 translateUninterpreted
-    :: ( Ord p
-       , p ~ TermLike variable
+    :: forall m variable
+     . ( Ord variable
        , Monad m
        )
     => SExpr  -- ^ type name
-    -> p  -- ^ uninterpreted pattern
-    -> Translator m p SExpr
-translateUninterpreted t pat | isVariable pat =
+    -> TranslateItem variable -- ^ uninterpreted pattern
+    -> Translator m variable SExpr
+translateUninterpreted _ (QuantifiedVariable _) =
+    error "Lemma encodings don't support quantified variables (yet)"
```

Comment:
Why not `empty` instead of `error`? That would match the existing behavior, which rejects quantifiers anyway.

Reply:
ok... I just thought we might want to find out if anyone tries to declare a lemma with quantifiers. If we use empty here, it would be silently ignored.